src/EmulatorCompat.sol

```diff
@@ -239,4 +284,34 @@ library UArchCompat {
     }
 
     function putChar(AccessLogs.Context memory a, uint8 c) internal pure {}
+
+    function uint32Log2(uint32 value) internal pure returns (uint32) {
+        require(value > 0, "EmulatorCompat: log2(0) is undefined");
+        return 31 - clz(value);
```

Comment:
I think we want this to be:

```
32 - clz(value - 1)
```

Which yields `ceil(log2(x))`. With this change, the implementation of `sendCmioResponse` can be simplified by removing the branch that increments `writeLengthLog2Size += 1;`.

Reply:
`clz(0)` is undefined (`__builtin_clz`).
The current implementation of `uint32Log2` is correct for all valid inputs.